app/validators/google_url_validator.rb

```diff
@@ -0,0 +1,30 @@
+class GoogleUrlValidator < ActiveModel::Validator
+  def validate(record)
+    uri = URI.parse(record.url)
+
+    validate_host(uri, record)
+    validate_path(uri, record)
+    validate_parameters(uri, record)
+  end
+
+private
+
+  def validate_host(uri, record)
+    unless uri.host == "docs.google.com"
+      record.errors[:url] << I18n.t('errors.invalid_hostname')
+    end
+  end
+
+  def validate_path(uri, record)
+    unless uri.path =~ %r{spreadsheets\/d\/.+\/pub}
+      record.errors[:url] << I18n.t('errors.invalid_path')
+    end
+  end
+
+  def validate_parameters(uri, record)
```

Comment:
Do you think it's worth validating somewhere that `output=tsv` in the params? The CSV parsing logic in `app/services/tag_importer/fetch_remote_data.rb` expects tab-separated values.